plugins/inputs/pgbouncer/pgbouncer.go

```diff
@@ -82,11 +168,16 @@ func (p *PgBouncer) Gather(acc telegraf.Accumulator) error {
 		acc.AddFields("pgbouncer", fields, tags)
 	}
 
-	err = rows.Err()
-	if err != nil {
-		return err
-	}
+	return rows.Err()
+}
 
+func (p *PgBouncer) showPools(acc telegraf.Accumulator) error {
+	// POOLS
+	var (
+		err     error
+		query   string
+		columns []string
+	)
```

Comment:
I know you are keeping things similar to how they used to be, but it would be nice to declare these things closer to where they are actually used. For example, creating query here, but then initializing it on the very next line is unnecessary extra code.

Same applies to showLists and showDatabases